### ISSUE TITLE
If a JN/NN/ZK task dies due to a slave being lost, it is never started.

### ISSUE THREAD

**Issue by abhay-agarwal** [mesosphere/hdfs#111]
This is because there are datanodes on every slave so there isn't any room to start another JN/NN/ZK when a slave is detached. We should instead be shutting down a datanode executor to make room.

![screen shot 2015-03-26 at 9 05 33 pm](https://cloud.githubusercontent.com/assets/673671/6861982/a56661fe-d3fc-11e4-9c61-4dcf9f09f227.png)

**Comment by elingg** [comment]
@abhay-agarwal , yes this will be fixed when there is a configurable number a DN's. A workaround is that when the slave dies, a new slave gets spin up and the JN or NN then starts since we move to the appropriate initialization state in the state machine. Also, see a known issue related to JN relocation, https://github.com/mesosphere/hdfs/issues/91
